app/scripts/storage/storage-file.js

```diff
@@ -15,82 +15,118 @@ const StorageFile = StorageBase.extend({
     load: function(path, opts, callback) {
         this.logger.debug('Load', path);
         const ts = this.logger.ts();
-        try {
-            const data = Launcher.readFile(path);
-            const rev = Launcher.statFile(path).mtime.getTime().toString();
-            this.logger.debug('Loaded', path, rev, this.logger.ts(ts));
-            if (callback) { callback(null, data.buffer, { rev: rev }); }
-        } catch (e) {
+
+        const onError = e => {
             this.logger.error('Error reading local file', path, e);
-            if (callback) { callback(e, null); }
-        }
+            if (callback) {
+                callback(e, null);
+            }
+        };
+
+        Launcher.readFile(path, data => {
+            Launcher.statFile(path, stat => {
+                const rev = stat.mtime.getTime().toString();
+                this.logger.debug('Loaded', path, rev, this.logger.ts(ts));
+                if (callback) {
+                    callback(null, data.buffer, { rev: rev });
+                }
+            }, onError);
+        }, onError);
     },
 
     stat: function(path, opts, callback) {
         this.logger.debug('Stat', path);
         const ts = this.logger.ts();
-        try {
-            const stat = Launcher.statFile(path);
+
+        Launcher.statFile(path, stat => {
             this.logger.debug('Stat done', path, this.logger.ts(ts));
-            if (callback) { callback(null, { rev: stat.mtime.getTime().toString() }); }
-        } catch (e) {
+            if (callback) {
+                const fileRev = stat.mtime.getTime().toString();
+                callback(null, { rev: fileRev });
+            }
+        }, e => {
             this.logger.error('Error stat local file', path, e);
-            if (e.code === 'ENOENT') {
-                e.notFound = true;
+            if (callback) {
+                callback(e, null);
             }
```

Comment:
Have you checked the code that relies on `notFound` property?
❗️ Restored the logic.